graphstorm-processing/docs/source/index.rst

```diff
@@ -0,0 +1,154 @@
+
+On MacOS you can install Java using ``brew``:
+
+.. code-block:: bash
+
+    brew install openjdk@11
+
+On Linux it will depend on your distribution's package
+manager. For Ubuntu you can use:
+
+.. code-block:: bash
+
+    sudo apt install openjdk-11-jdk
+
+On Amazon Linux 2 you can use:
+
+.. code-block:: bash
+
+    sudo yum install java-11-amazon-corretto-headless
+    sudo yum install java-11-amazon-corretto-devel
+
+To check if Java is installed you can use.
+
+.. code-block:: bash
+
+    java -version
+
+
+Example
+-------
+
+See the provided :doc:`usage/example` for an example of how to start with tabular
+data and convert them into a graph representation before partitioning and
+training with GraphStorm.
+
+Usage
+-----
+
+To use the library to process your data, you will need to have your data
+in a tabular format, and a corresponding JSON configuration file that describes the
+data. The input data can be in CSV (with header(s)) or Parquet format.
+
+The configuration file can be in GraphStorm's GConstruct format,
+with the caveat that the file paths need to be relative to the
+location of the config file. See :doc:`/usage/example` for more details.
+
+After installing the library, executing a processing job locally can be done using:
+
+.. code-block:: bash
+
+    gs-processing \
+        --config-filename gconstruct-config.json \
+        --input-prefix /path/to/input/data \
+        --output-prefix /path/to/output/data
+
+Once the processing engine has processed the data, we want to ensure
+they match the requirements of the DGL distributed partitioning
+pipeline, so we need to run an additional script that will
+make sure the produced data matches the assumptions of DGL [#f1]_.
+
+.. note::
+
+    Ensure you pass the output path of the previous step as the input path here.
+
+.. code-block:: bash
+
+    gs-repartition \
+        --input-prefix /path/to/output/data
+
+Once this script completes, the data are ready to be fed into DGL's distributed
+partitioning pipeline.
+See `this guide <https://github.com/awslabs/graphstorm/blob/main/sagemaker/README.md#launch-graph-partitioning-task>`_
+for more details on how to use GraphStorm distributed partitioning on SageMaker.
+
+See :doc:`/usage/example` for a detailed walkthrough of using GSProcessing to
+wrangle data into a format that's ready to be consumed by the GraphStorm/DGL
+partitioning pipeline.
+
+
+Using with Amazon SageMaker
+---------------------------
+
+To run distributed jobs on Amazon SageMaker we will have to build a Docker image
+and push it to the Amazon Elastic Container Registry, which we cover in
+:doc:`usage/distributed-processing-setup` and run a SageMaker Processing
+job which we describe in :doc:`/usage/amazon-sagemaker`.
+
+
+Developer guide
+---------------
+
+To get started with developing the package refer to :doc:`/developer/developer-guide`.
+
+
+.. rubric:: Footnotes
+
+.. [#f1] DGL expects that every file produced for a single node/edge type
```

Comment:
Do we really want users to know this?

Reply:
I think yes, because it's not documented anywhere else.